db/db_wal_test.cc

```diff
@@ -88,6 +88,7 @@ class DBWALTestWithEnrichedEnv : public DBTestBase {
     enriched_env_ = new EnrichedSpecialEnv(env_->target());
     auto options = CurrentOptions();
     options.env = enriched_env_;
+    options.allow_2pc = true;
```

Comment:
Why is it added?

Reply:
[On line 535 in `dp_impl_open.cc`](https://github.com/facebook/rocksdb/blob/d59549298fcd50e4aaa59af3dfc039d9a4db5623/db/db_impl_open.cc#L535) we skip WAL files based on `versions_->min_log_number_to_keep_2pc()`.

If I am correct, `min_log_number_to_keep_2pc_` is set only in 2PC mode.

See comment [on line 2976 in `version_set.cc`](https://github.com/facebook/rocksdb/blob/d59549298fcd50e4aaa59af3dfc039d9a4db5623/db/version_set.cc#L2976) .


[On line 3926 in `version_set.cc`](https://github.com/facebook/rocksdb/blob/d59549298fcd50e4aaa59af3dfc039d9a4db5623/db/version_set.cc#L), it is set as `version_edit::min_log_number_to_keep_`, which is set only if `options.allow_2pc` is true.

See [line 389 in `memtable_list.cc`](https://github.com/facebook/rocksdb/blob/d59549298fcd50e4aaa59af3dfc039d9a4db5623/db/memtable_list.cc#L389).

So if we want to "SkipDeletedWALs", we have to enable `options.allow_2pc`.

Please correct me if I am wrong.

Thank you.

Reply:
But how does it related to your change?

Reply:
`SkipDeletedWALs` should fail because `options.allow_2pc` is not enabled.
It passed before only because there is a extra '/' in the name of WAL file when it is deleted.

For example, when `EnrichedSpecialEnv::DeleteFile()` is called, the file name is `testrocksdb-26040/xxx_2273688893793634580//000003.log`.
When `EnrichedSpecialEnv::NewSequentialFile()` is called, the file name is `testrocksdb-26040/xxx_2273688893793634580/000003.log `.

The same file has two different names, the first one has an extra '/'.
That's why [line 31 in `db_wal_test.cc `](https://github.com/facebook/rocksdb/blob/d59549298fcd50e4aaa59af3dfc039d9a4db5623/db/db_wal_test.cc#L31) is not hit.

`SkipDeletedWALs ` passed "by mistake".

After I fixed the extra '/' issue for `FaultInjectionTestSplitted.FaultTest`, this test failed.
Then I found we have to enable `options.allow_2pc` for this test.

Reply:
Let me take a closer look. We should handle log gap correctly no matter 2pc is on or off.

Reply:
You are right. This is not supposed to pass. Thank you for fixing it.